From 29d9a8b1e5266a3fc645b1cb9ea4617abbb09c82 Mon Sep 17 00:00:00 2001
From: Stefan Bodewig <bodewig@apache.org>
Date: Mon, 3 Jun 2013 09:58:27 +0000
Subject: [PATCH] COMPRESS-229 aftermaths, properly deal with broken archives
 that end with an incomplete entry.  Unfortunately we don't have a test for
 this

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/compress/trunk@1488947 13f79535-47bb-0310-9956-ffa450edef68
---
 .../archivers/tar/TarArchiveInputStream.java         | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveInputStream.java b/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveInputStream.java
index 1eb0b6839ab..c52db9a1760 100644
--- a/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveInputStream.java
+++ b/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveInputStream.java
@@ -250,11 +250,23 @@ public TarArchiveEntry getNextTarEntry() throws IOException {
 
         if (currEntry.isGNULongLinkEntry()) {
             byte[] longLinkData = getLongNameData();
+            if (longLinkData == null) {
+                // Bugzilla: 40334
+                // Malformed tar file - long link entry name not followed by
+                // entry
+                return null;
+            }
             currEntry.setLinkName(encoding.decode(longLinkData));
         }
 
         if (currEntry.isGNULongNameEntry()) {
             byte[] longNameData = getLongNameData();
+            if (longNameData == null) {
+                // Bugzilla: 40334
+                // Malformed tar file - long entry name not followed by
+                // entry
+                return null;
+            }
             currEntry.setName(encoding.decode(longNameData));
         }
 
